src/distilabel/tasks/text_generation/evol_instruct.py

```diff
@@ -0,0 +1,155 @@
+    The In-breadth Evolving is mutation, i.e., generating a completely new instruction based on the given instruction.
+
+    Given the evolved instructions are generated from LLMs, sometimes the evolving will fail. We adopt an instruction eliminator
+    to filter the failed instructions, called Elimination Evolving, but we don't apply the step of asking again to the LLM it the
+    answer is a copy from the same used prompt.
+
+    This evolutionary process can be repeated for several rounds to obtain instruction data containing various complexities.
+    Currently the task is implemented as a single step, so to generate multiple evolutions you can "repeat" the instructions
+    in the original dataset. An example can be seen at the following script:
+    [examples/pipeline-evol-instruct-alpaca.py](https://github.com/argilla-io/distilabel/tree/main/examples/pipeline-evol-instruct-alpaca.py)
+
+    Args:
+        system_prompt (str, optional): the system prompt to be used. Not defined for this task.
+
+    References:
+        - [`WizardLM: Empowering Large Language Models to Follow Complex Instructions`](https://arxiv.org/pdf/2304.12244.pdf)
+    """
+
+    system_prompt: str = ""
+
+    __jinja2_template__: str = _EVOL_INSTRUCT_TEMPLATE
+
+    def generate_prompt(
+        self, input: str, evolution_method: Optional[EvolutionMethod] = None, **_: Any
+    ) -> Prompt:
+        """Generates a prompt following the Evol-Instruct specification.
+
+        Args:
+            input (str): the input to be used for the prompt.
+            evolution_method (str, optional): The evolution method to be used. If not provided (the default), a random one is chosen
+                like the original paper. Available ones are "breadth", "constraints", "deepen", "concretizing" and "reasoning".
+
+        Returns:
+            Prompt: the generated prompt.
+
+        Examples:
+            >>> from distilabel.tasks.text_generation import EvolInstructTask
+            >>> task = EvolInstructTask()
+            >>> task.generate_prompt("Give three tips for staying healthy.")
+            Prompt(
+                system_prompt="",
+                formatted_prompt="I want you to act as a Prompt ...",
+            )
+        """
+        if not evolution_method:
+            evolution_method = random.choice(get_args(EvolutionMethod))
+
+        render_kwargs = {
+            "evol_method": evolution_method,
+            "instruction": input,
+        }
+        return Prompt(
+            system_prompt=self.system_prompt,
+            formatted_prompt=self.template.render(**render_kwargs),
+        )
+
+    @property
+    def output_args_names(self) -> List[str]:
+        return ["instruction"]
+
+    def _elimination_evolving(self, output: str) -> Optional[str]:
+        """Performs the elimination step of the Evol-Instruct task, steps 2-4 in the paper:
+
```

Comment:
Maybe we should still outline step 1? and add something [NOT IMPLEMETED] 1. blabla bla?